### PR TITLE
[G2M] Plotly - state callback when plot is loaded

### LINKED ISSUE #206: Plotly - add `after plot` event
**Problem:**

- the Plotly charts need to send a signal to the parent comp that they finished rendering
- this is used in snake-dashboard to enable PDF download only when all charts are rendered

[Reference](https://plotly.com/javascript/plotlyjs-events/#afterplot-event:~:text=it%27s%20WebGL%20context.-,Afterplot%20Event,-The%20event%20handler)

**Solution:**
- add a callback to the prop list of the Plotly charts that would be able to update render chart status in the parent component

### PR DESCRIPTION
**Pull request by kc-leung**
fixes #206 

- Added function in promise when plot is loaded

Tested in widget studio & dashboard. Since there are multiple widgets in each report on the dashboard side a flag or counter will need to be added to match the number of widget present in the report to tell all the widgets are loaded

### CODE REVIEW

File: src/components/plotly/shared/plot.js [hunk 1]
```diff
@@ -5,34 +5,45 @@ import Plotly from 'plotly.js-basic-dist-min'
...
     try {
       if (data && layout && config) {
-        ref && Plotly.react(ref, { data, layout, config }).catch(() => { })
+        ref && Plotly.react(ref, { data, layout, config })
+          .then((res) => { 
+            onAfterPlot({ response: res, isLoaded: true })
+          })
+          .catch((err) => {
+            onAfterPlot({ response: err, isLoaded: false })
+          })
```
Comment (woozyking) [inline src/components/plotly/shared/plot.js:23]: Not new with this change, but how does this work with promise and synchronous try/catch together? Seems a bit weird and not as coherent

Also prefer using `if` than `&&` to carry out the logic execution (unless in JSX context)

**Review by woozyking — APPROVED** [review]

File: src/components/plotly/shared/plot.js (hunk 1 above)
Comment (kc-leung) [reply src/components/plotly/shared/plot.js:23]: It is a promise from plotly side which returns the Ref of the graph when it is loaded in the DOM.

**Review by Hamaad102 — APPROVED** [review]
Looks good, thank you!

File: src/components/plotly/shared/plot.js (hunk 1 above)
Comment (geoerika) [reply src/components/plotly/shared/plot.js:23]: Agree with Leo, include ref into the above if conditions.

**Review by geoerika — APPROVED** [review]

**Pull request merged**